components/channel_header/channel_header.js

```diff
@@ -116,12 +114,17 @@ export default class ChannelHeader extends React.PureComponent {
         if (header !== prevHeader) {
             this.props.actions.getCustomEmojisInText(header);
         }
+        if (this.props.rhsOpen !== prevProps.rhsOpen) {
+            this.handleResize();
```

Comment:
This should be done using `getDerivedStateFromProps` since it updates the state of the component. Either that or we can leave `this.state.showSearchBar` as it was before where it only depends on the size of the window and check `this.state.showSearchBar || this.props.rhsOpen` in the render method